src/consensus/aft/orc_requests.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/thread_messaging.h"
+#include "kv/kv_types.h"
+#include "node/node_client.h"
+#include "node/rpc/serdes.h"
+
+namespace ccf
+{
+  struct ObservedReconfigurationCommit
+  {
+    struct In
+    {
+      ccf::NodeId from;
+      kv::ReconfigurationId reconfiguration_id;
+    };
+
+    using Out = void;
+  };
+
+  DECLARE_JSON_TYPE(ObservedReconfigurationCommit::In)
+  DECLARE_JSON_REQUIRED_FIELDS(
+    ObservedReconfigurationCommit::In, from, reconfiguration_id)
+
+  inline bool submit_orc(
+    std::shared_ptr<ccf::NodeClient> client,
+    const ccf::NodeId& from,
+    kv::ReconfigurationId rid)
+  {
+    LOG_DEBUG_FMT("Configurations: submit ORC for #{} from {}", rid, from);
+
+    ObservedReconfigurationCommit::In ps = {from, rid};
+
+    http::Request request(fmt::format(
+      "/{}/{}", ccf::get_actor_prefix(ccf::ActorsType::nodes), "orc"));
+    request.set_header(
+      http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
+
+    auto body = serdes::pack(ps, serdes::Pack::Text);
+    request.set_body(&body);
+    return client->make_request(request);
```

Comment:
It took me a while to wrap my head around this: this is actually sending a request to itself and relying on the forwarding mechanism in the generic frontend to send the request to the new primary. I think this is OK. However, the logic in `http_node_client.h` does not actually wait for the response to the request (it will take a while to receive it from the forwardee) and I think we should wait for this asynchronously, and also wait for commit on this transaction.

Reply:
In general that is true (and I've started adding code for that in the split identity PR), but for ORCs it's not necessary to wait. The "reply" is essential the primary changing the node status in the nodes table. I believe, if necessary, all nodes will hit the same reconfiguration commit check in `raft.h` again and re-schedule new ORCs.